### PR TITLE
Automation filter components [MAILPOET-4624] [MAILPOET-5001] [MAILPOET-5187]

### PR DESCRIPTION
**Pull request by JanJakes**
## Description

Implements number, integer, enum, and boolean filters.

## Code review notes

The pull request is against https://github.com/mailpoet/mailpoet/pull/4828, which needs to be merged first. Please, if approved, assign back to me, I'll take care of the rebase.

## QA notes

_N/A_

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/714

## Linked tickets

[MAILPOET-4624]
[MAILPOET-5001]
[MAILPOET-5187]

[MAILPOET-4624]: https://mailpoet.atlassian.net/browse/MAILPOET-4624?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MAILPOET-5001]: https://mailpoet.atlassian.net/browse/MAILPOET-5001?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MAILPOET-5187]: https://mailpoet.atlassian.net/browse/MAILPOET-5187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/Automation/Integrations/Core/Filters/IntegerFilter.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    if (!$matches) {
+      return false;
+    }
+
+    if (isset($value) && !$this->isWholeNumber($value)) {
+      return false;
+    }
+
+    $filterValue = $data->getArgs()['value'] ?? null;
+    if (is_array($filterValue)) {
```
Comment (websupporter) [inline mailpoet/lib/Automation/Integrations/Core/Filters/IntegerFilter.php:35]: If I understand this correctly `$filterValue` can't be an array as `parent::matches()` would return `false` in the case `$filterValue` is not a number.

Comment (websupporter) [reply mailpoet/lib/Automation/Integrations/Core/Filters/IntegerFilter.php:35]: Ah! `matchesBetween()` returns early in the `parent`. Ok, my fault.

File: mailpoet/lib/Automation/Integrations/MailPoet/Subjects/SubscriberSubject.php [hunk 2]
```diff
@@ -94,14 +101,25 @@ function (SubscriberPayload $payload) {
         },
         [
           'options' => [
-            SubscriberEntity::STATUS_SUBSCRIBED => __('Subscribed', 'mailpoet'),
-            SubscriberEntity::STATUS_UNCONFIRMED => __('Unconfirmed', 'mailpoet'),
-            SubscriberEntity::STATUS_UNSUBSCRIBED => __('Unsubscribed', 'mailpoet'),
-            SubscriberEntity::STATUS_BOUNCED => __('Bounced', 'mailpoet'),
+            [
```
Comment (websupporter) [inline mailpoet/lib/Automation/Integrations/MailPoet/Subjects/SubscriberSubject.php:104]: `STATUS_INACTIVE` seems missing.

File: mailpoet/tests/unit/Automation/Integration/Core/Filters/EnumArrayFilterTest.php [hunk 3]
```diff
@@ -88,8 +88,8 @@ public function testMatchesNoneCondition(): void {
   }
 
   public function testUnknownCondition(): void {
-    $this->assertNotMatches('unknown', '', '');
-    $this->assertNotMatches('unknown', 'abc', 'abc');
+    $this->assertNotMatches('unknown', [1], [1]);
+    $this->assertNotMatches('unknown', [1], [1]);
```
Comment (websupporter) [inline mailpoet/tests/unit/Automation/Integration/Core/Filters/EnumArrayFilterTest.php]: One time should be enough ;)

**Review by websupporter — DISMISSED** [review]
Awesome @JanJakes . Too minor remarks, but good to go imo.

**Comment by JanJakes** [comment]
@websupporter Thanks, fixed!

**Review by websupporter — APPROVED** [review]
lgtm

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**